### PR TITLE
Replace boilerplate serialization

### PR DESCRIPTION
**Pull request by oxade**
@huitseeker 
Can you provide a descriptive comment for the method.
Also we probably need a better name for it.

### CODE REVIEW

**Review by asonnino — APPROVED** [review]
LGTM

If I want to use the function `skip_prior_to` of the store with a key of type `(TransactionDigest, SequenceNumber)`, is there a way to tell the store to "sort" those keys by sequence number rather than a weird mix of TransactionDigest and SequenceNumber?

File: typed-store/src/rocks/mod.rs [hunk 1]
```diff
@@ -451,3 +425,16 @@ pub fn open_cf<P: AsRef<Path>>(
     };
     Ok(rocksdb)
 }
+
+/// TODO: Good description of why we're doing this
+#[inline(always)]
```
Comment (huitseeker) [inline typed-store/src/rocks/mod.rs]: Please replace with `#[inline]`, the always is not the best idea here, even though it sounds like it should:
http://troubles.md/rust-optimization/#dont-use-inline-always

File: typed-store/src/rocks/mod.rs [hunk 2]
```diff
@@ -451,3 +425,16 @@ pub fn open_cf<P: AsRef<Path>>(
...
+pub fn be_fix_int_ser<S>(t: &S) -> Result<Vec<u8>, TypedStoreError>
```
Comment (huitseeker) [inline typed-store/src/rocks/mod.rs:431]: How about `serialize_big_endian`, or something more evocative?

File: typed-store/src/rocks/mod.rs [hunk 3]
```diff
@@ -451,3 +425,16 @@ pub fn open_cf<P: AsRef<Path>>(
     };
     Ok(rocksdb)
 }
+
+/// TODO: Good description of why we're doing this
```
Comment (huitseeker) [inline typed-store/src/rocks/mod.rs]: The doc on the default Rocks comparator being BE is a bit obscure, and you'd have to look into rust-rocksdb for how this is selected every time, but we can certainly mention in a sentence that "rocksdb stores keys in BE and has a seek operator on iterators": 
https://github.com/facebook/rocksdb/wiki/Iterator#introduction

**Review by huitseeker — CHANGES_REQUESTED** [review]
Thanks, this is much cleaner.
The `#[inline(always)]` will hit compilation times (which are already not super) hard, and the ThinLTO can inline most cases already so I think we should get rid of the always. I left a few other remarks.

**Review by huitseeker — APPROVED** [review]

**Pull request merged**